crates/remora/src/executor/fake.rs

```diff
@@ -0,0 +1,196 @@
+            updates: TransactionEffects::default(),
+            new_state: BTreeMap::new(),
+        }
+    }
+}
+
+impl ExecutableTransaction for FakeTransaction {
+    fn digest(&self) -> &TransactionDigest {
+        // Return a reference to a static TransactionDigest
+        &TransactionDigest::ZERO
+    }
+
+    fn input_objects(&self) -> Vec<InputObjectKind> {
+        // Return an empty vector since the method won't be used
+        Vec::new()
+    }
+}
+
+impl ObjectStore for FakeStore {
+    fn get_object(
+        &self,
+        _object_id: &ObjectID,
+    ) -> Result<Option<Object>, sui_types::storage::error::Error> {
+        unreachable!()
+    }
+
+    fn get_object_by_key(
+        &self,
+        _object_id: &ObjectID,
+        _version: VersionNumber,
+    ) -> Result<Option<Object>, sui_types::storage::error::Error> {
+        unreachable!()
+    }
+}
+
+impl ParentSync for FakeStore {
+    fn get_latest_parent_entry_ref_deprecated(
+        &self,
+        _object_id: ObjectID,
+    ) -> SuiResult<Option<ObjectRef>> {
+        unreachable!()
+    }
+}
+
+impl ChildObjectResolver for FakeStore {
+    fn read_child_object(
+        &self,
+        parent: &ObjectID,
+        child: &ObjectID,
+        child_version_upper_bound: SequenceNumber,
+    ) -> SuiResult<Option<Object>> {
+        Ok(self.get_object(child).unwrap().and_then(|o| {
+            if o.version() <= child_version_upper_bound
+                && o.owner == Owner::ObjectOwner((*parent).into())
+            {
+                Some(o.clone())
+            } else {
+                None
+            }
+        }))
+    }
+
+    fn get_object_received_at_version(
+        &self,
+        _owner: &ObjectID,
+        _receiving_object_id: &ObjectID,
+        _receive_object_at_version: SequenceNumber,
+        _epoch_id: EpochId,
+    ) -> SuiResult<Option<Object>> {
+        unimplemented!()
+    }
+}
+
+impl BackingPackageStore for FakeStore {
+    fn get_package_object(&self, _package_id: &ObjectID) -> SuiResult<Option<PackageObject>> {
+        unreachable!()
+    }
+}
+
+impl StateStore<TransactionEffects> for FakeStore {
+    fn commit_objects(&self, _updates: TransactionEffects, _new_state: BTreeMap<ObjectID, Object>) {}
+}
+
+#[derive(Clone)]
+pub struct FakeExecutor;
+
+impl Executor for FakeExecutor {
+    type Transaction = FakeTransaction;
+    type ExecutionResults = TransactionEffects;
+    type Store = FakeStore;
+
+    fn context(&self) -> Option<std::sync::Arc<BenchmarkContext>> {
+        None
+    }
+
+    async fn execute(
+        _ctx: Option<std::sync::Arc<BenchmarkContext>>,
+        _store: std::sync::Arc<Self::Store>,
+        transaction: &FakeTransactionType,
+    ) -> FakeExecutionResults {
+
+        let start = Instant::now();
+        let duration = Duration::from_millis(transaction.delay);
+    
+        // Busy-wait until the specified duration has elapsed.
+        while Instant::now() - start < duration {}
```

Comment:
mmh it seems that the compiler optimises away this line

Reply:
Here's my [code](https://gist.github.com/Scofield626/48bc1926dbd22a197573d9d7b0142ff6) for having a calibrated iterations of fake works according to the txn time we need. It works out well for my previous experiments. The only nitty thing is on line 14, which you may need a large number for txn runtime longer than 10ms.